### ISSUE TITLE
elementary OS is detected as Debian rather than Ubuntu

### ISSUE THREAD

**Issue by aronbohl** [rvm/rvm#3935]
## Description

On a fresh install of elementary OS, the system is detected as Debian rather than as an Ubuntu derivative causing requirement issues when installing a ruby version less than 2.4. I'm not sure if this is a problem with rvm's detection or the way this distribution identifies itself.

## Steps to reproduce

 1. Install elementary OS
 2. Install rvm (\curl -sSL https://get.rvm.io | bash -s stable --rails)
 3. Install a Ruby version less than 2.4 (I was installing 2.2.6)

## Actual behavior
Compiling fails with the following error because it is a package requirement for Debian but not Ubuntu:

Searching for binary rubies, this might take some time.
No binary rubies available for: debian/9/x86_64/ruby-2.3.2.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for debian.
Installing requirements for debian.
Updating system defaultuser password required for 'apt-get --quiet --yes update': ..         
...
Installing required packages: libssl1.0-dev....
Error running 'requirements_debian_libs_install libssl1.0-dev',
showing last 15 lines of /home/defaultuser/.rvm/log/1487881738_ruby-2.3.2/package_install_libssl1.0-dev.log
++ [[ -d /usr/local/sbin ]]
++ [[ :/home/defaultuser/.rvm/gems/ruby-2.3.1/bin:/home/defaultuser/.rvm/gems/ruby-2.3.1@global/bin:/home/defaultuser/.rvm/rubies/ruby-2.3.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/defaultuser/apps/AWS-ElasticBeanstalk-CLI-2.6.4/eb/linux/python2.7:/home/defaultuser/.rvm/bin:/home/defaultuser/apps/AWS-ElasticBeanstalk-CLI-2.6.4/eb/linux/python2.7:/home/defaultuser/.rvm/bin:/home/defaultuser/apps/AWS-ElasticBeanstalk-CLI-2.6.4/eb/linux/python2.7:/home/defaultuser/.rvm/bin: != *\:\/\u\s\r\/\l\o\c\a\l\/\s\b\i\n\:* ]]
++ [[ -n '' ]]
++ command_to_run=(__rvm_sudo -p "%p password required for '$*': " "${command_to_run[@]}")
++ __rvm_sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
++ command sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
++ sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libssl1.0-dev
E: Couldn't find any package by glob 'libssl1.0-dev'
E: Couldn't find any package by regex 'libssl1.0-dev'
++ return 100
++ return 100
Requirements installation failed with status: 100.

## RVM Info

ruby-2.3.1:

  system:
    uname:        "Linux Precision-5510 4.8.0-39-generic #42~16.04.1-Ubuntu SMP Mon Feb 20 15:06:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux"
    name:         "Debian"
    version:      "9"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 4.3.46(1)-release (x86_64-pc-linux-gnu)"
    zsh:          " => not installed"
    remote_path:  "debian/9/x86_64"

  rvm:
    version:      "rvm 1.29.1 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io/]"
    updated:      "1 day 4 hours 51 minutes 53 seconds ago"
    path:         "/home/defaultuser/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "2.3.1p112"
    date:         "2016-04-26"
    platform:     "x86_64-linux"
    patchlevel:   "2016-04-26 revision 54768"
    full_version: "ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]"

I realize it's impractical to code for every Linux distro out there but I just wanted to see how feasible a fix was. Also, if the distro is identifying itself improperly, I'll file a bug over there. As a work around I've downloaded a copy of /etc/lsb-release from Ubuntu 16.04 to a temp directory and pointed the detect_system script at that. Everything then installs as expected. Let me know if I can provide any more info.

**Comment by pkuczynski** [comment]
Could you please list output of following commands:

```
cat /etc/lsb-release
cat /etc/debian_version
cat /etc/os-release
dpkg --print-architecture
```

**Comment by aronbohl** [comment]
Sorry, I should have included that before. Here's the info:

```
$ cat /etc/lsb-release
DISTRIB_ID="elementary"
DISTRIB_RELEASE=0.4
DISTRIB_CODENAME=loki
DISTRIB_DESCRIPTION="elementary OS 0.4 Loki"

$ cat /etc/debian_version
stretch/sid

$ cat /etc/os-release
NAME="elementary OS"
VERSION="0.4 Loki"
ID="elementary"
ID_LIKE=ubuntu
PRETTY_NAME="elementary OS 0.4 Loki"
VERSION_ID="0.4"
HOME_URL="http://elementary.io/"
SUPPORT_URL="http://elementary.io/support/"
BUG_REPORT_URL="https://bugs.launchpad.net/elementary/+filebug"
VERSION_CODENAME=loki
UBUNTU_CODENAME=loki

$ dpkg --print-architecture
amd64
```

**Comment by pkuczynski** [comment]
Can you please try:

```
rvm get branch /feature/3935-elementary-os
rvm install 2.2.6
```

**Comment by aronbohl** [comment]
It still detects Debian but I think I see the problem. Your `.?` isn't getting interpreted as a regular expression. I think your `GREP_OPTIONS` just needs a `-E`.

`GREP_OPTIONS="-E" \command \grep "DISTRIB_ID=.?elementary" /etc/lsb-release >/dev/null`

I made the change and then everything worked fine and it installed 2.2.6.

**Comment by pkuczynski** [comment]
Indeed. Fixed that and will merge after review from @mpapis

**Comment by aronbohl** [comment]
Thanks! I appreciate the help.

**Comment by aronbohl** [comment]
One more thing that I just noticed, adding something to `GREP_OPTIONS` throws a warning:

`grep: warning: GREP_OPTIONS is deprecated; please use an alias or script`

Maybe use egrep instead?

**Comment by mpapis** [comment]
@aronbohl does it show only once? Or few in a group on start?

**Comment by aronbohl** [comment]
@mpapis it was showing in clusters of 3 every time I'd run something like `rvm install` or `rvm info`. Although, I just switched back to the master branch after this got merged in and I don't see the warning anymore so it must be resolved.

**Comment by hhalmeida** [comment]
Hi, I have this problem, when install ruby-2.3.3, I use linux ElementaryOS (Loki), I tried to install this
```
rvm get branch /feature/3935-elementary-os
rvm install 2.3
```
... but it says it does not exist anymore, where I find it to install, or what would be another way to solve it, I'm with Ruby 2.4 installed.

my error ::: 
```
hugo@hhalmeida:~$ rvm install 2.3
Warning! PATH is not properly set up, '/home/hugo/.rvm/gems/ruby-2.4.0/bin' is not at first place.
         Usually this is caused by shell initialization files. Search for 'PATH=...' entries.
         You can also re-add RVM to your profile by running: 'rvm get stable --auto-dotfiles'.
         To fix it temporarily in this shell session run: 'rvm use ruby-2.4.0'.
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
Searching for binary rubies, this might take some time.
No binary rubies available for: debian/9/x86_64/ruby-2.3.3.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for debian.
Installing requirements for debian.
Updating system.....
Installing required packages: libssl1.0-dev....
Error running 'requirements_debian_libs_install libssl1.0-dev',
showing last 15 lines of /home/hugo/.rvm/log/1491306380_ruby-2.3.3/package_install_libssl1.0-dev.log
++ [[ -d /usr/local/sbin ]]
++ [[ :/home/hugo/.dropbox-bin:/home/hugo/.rvm/gems/ruby-2.4.0/bin:/home/hugo/.rvm/gems/ruby-2.4.0@global/bin:/home/hugo/.rvm/rubies/ruby-2.4.0/bin:/home/hugo/bin:/home/hugo/.local/bin:/home/hugo/.nvm/versions/node/v6.10.0/bin:/home/hugo/.dropbox-bin:/home/hugo/bin:/home/hugo/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/jvm/java-8-oracle/bin:/usr/lib/jvm/java-8-oracle/db/bin:/usr/lib/jvm/java-8-oracle/jre/bin:/home/hugo/.rvm/bin:/usr/lib/jvm/java-8-oracle/bin:/usr/lib/jvm/java-8-oracle/db/bin:/usr/lib/jvm/java-8-oracle/jre/bin:/home/hugo/.rvm/bin:/home/hugo/.rvm/bin: != *\:\/\u\s\r\/\l\o\c\a\l\/\s\b\i\n\:* ]]
++ [[ -n '' ]]
++ command_to_run=(__rvm_sudo -p "%p password required for '$*': " "${command_to_run[@]}")
++ __rvm_sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
++ command sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
++ sudo -p '%p password required for '\''apt-get --no-install-recommends --yes install libssl1.0-dev'\'': ' apt-get --no-install-recommends --yes install libssl1.0-dev
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libssl1.0-dev
E: Couldn't find any package by glob 'libssl1.0-dev'
E: Couldn't find any package by regex 'libssl1.0-dev'
++ return 100
++ return 100
Requirements installation failed with status: 100.
```

Best Regards

**Comment by mpapis** [comment]
@hhalmeida please try with `rvm get master` and if it does not work please open a new issue - this one is already solved.

**Comment by darkn3rd** [comment]
I have this issue with zorion os and cannot get ruby-2.3 on my system.  I keep getting `Couldn't find any package by glob 'libssl1.0-dev'`

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Zorin
Description:	Zorin OS 12.1
Release:	12
Codename:	xenial
```

**Comment by tex87** [comment]
Hi,
I installed rvm on elementary os loki and when I tried to install ruby-2.3.3 I get the same error as @aronbohl .
In which file can I add the string GREP_OPTIONS="-E" \command \grep "DISTRIB_ID=.?elementary" /etc/lsb-release >/dev/null ?
I even tried "rvm get master" but I get a "* WARNING: you have GEM_HOME="/home/myname/.rvm/gems/ruby-2.4.0" this is conflicting with RVM, make sure to:
unset GEM_HOME"

thanks.

**Comment by kevyder** [comment]
Hi,  i fixed running the command:
`rvm autolibs disabled`

**Comment by Casual3498** [comment]
```
I was unable to get branch, but 'rvm autolibs disabled' helped me.

rvm get branch /feature/3935-elementary-os
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/rvm/rvm/master/binscripts/rvm-installer.asc
Verifying /home/serge/.rvm/archives/rvm-installer.asc
gpg: Подпись сделана Вс 02 июл 2017 23:42:38 +03 ключом RSA с ID BF04FF17
gpg: Действительная подпись пользователя "Michal Papis (RVM signing) <mpapis@gmail.com>" [неизвестно]
gpg:                 или "Michal Papis <michal.papis@toptal.com>" [неизвестно]
gpg:                 или "[jpeg image of size 5015]" [неизвестно]
gpg: Внимание: Данный ключ не заверен доверенной подписью!
gpg:          Нет указаний на то, что подпись принадлежит владельцу.
Отпечаток главного ключа: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Отпечаток подключа: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/serge/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/feature/3935-elementary-os.tar.gz
curl: (22) The requested URL returned error: 404 Not Found

Could not download 'https://github.com/rvm/rvm/archive/feature/3935-elementary-os.tar.gz'.
  curl returned status '22'.

Downloading https://bitbucket.org/mpapis/rvm/get/feature/3935-elementary-os.tar.gz
curl: (22) The requested URL returned error: 404 Not Found

Could not download 'https://bitbucket.org/mpapis/rvm/get/feature/3935-elementary-os.tar.gz'.
  curl returned status '22'.

Could not update RVM, get some help at #rvm IRC channel at freenode servers.

```

**Comment by mpapis** [comment]
@Casual3498 please try with rvm get master and if it does not work please open a new issue - this one is already solved.

**Comment by darkn3rd** [comment]
This is broken in Zorin OS, detected as Debian, when it is Ubuntu.

**Comment by Casual3498** [comment]
@mpapis I installed ruby 2.1 with 'rvm autolibs disabled'.
Now I type  (for testing) 'rvm get master', then 'rvm autolibs enabled' and 'rvm install 2.3'. All is worked.

**Comment by gkalil** [comment]
When i do rvm get master i get a bunch of Cannot utime: Operation not permitted and end with Could not update RVM, please report to https://github.com/rvm/rvm/issues.
